File: lib/typescript/src/lib/ufdata.ts
import { Project } from './project';

/**
 * UtaFormatix Data Document.
 * @param formatVersion - The version of the format. Current version is [UtaFormatixDataVersion].
 * @param project - Content of the project.
 */
export type UfData = {
  formatVersion: number;
  project: Project;
};

File: lib/csharp/UtaFormatix.Data/UfData.cs
namespace UtaFormatix.Data
{

    /// <summary>UtaFormatix Data Document.</summary>
    ///
    /// <param name="FormatVersion">The format version of this document.</param>
    /// <param name="Project">Content of the project.</param>
    public record UfData(int FormatVersion, Project Project)
    {
        /// <summary>
        /// Current format version implemented by this library.
        /// </summary>
        public const int CurrentFormatVersion = 1;
    }
}
